src/inc/msquic.h

```diff
@@ -1255,8 +1255,9 @@ typedef struct QUIC_STREAM_EVENT {
         struct {
             BOOLEAN ConnectionShutdown;
             BOOLEAN AppCloseInProgress       : 1;
-            BOOLEAN ConnectionShutdownByPeer : 1;
-            BOOLEAN RESERVED                 : 6;
+            BOOLEAN ConnectionShutdownByApp  : 1;
+            BOOLEAN ConnectionClosedRemotely : 1;
```

Comment:
I'm fine with this change, but you definitely will want to make sure you add some tests to verify things operate as you expect. For instance, I'm not 100% positive that `Connection->State.ClosedRemotely` will always give you what you want. That flag doesn't mean the remote closed first, just that the remote side has closed (which can sometimes happen **implicitly**, such as with idle timeouts, I believe).

Reply:
Ok. Feel free to leverage existing tests to add additional logic just to verify these flags.